python/ray/serve/_private/proxy.py

```diff
@@ -678,7 +686,13 @@ async def unary_unary(
                 stream=False,
             )
             proxy_response = await self.proxy_request(proxy_request=proxy_request)
-            return proxy_response.response
+            if proxy_response.streaming_response is not None:
+                # Unary calls go through the same generator codepath but will only ever
+                # yield a single result.
+                async for result in proxy_response.streaming_response:
+                    return result
+            else:
+                return proxy_response.response
```

Comment:
I feel since the response object for both the unary and streaming case are now merged, we don't really need this additional case and the response attribute on the proxy response. But I can clean it my in a follow up. Thanks for the refactor!

Reply:
This is still used for the "not found," "timeout," etc.

We can definitely fully unify them, just minimizing changes here.

Reply:
Agreed, I can do a clean up afterwards.